### DISCUSSION TITLE
include via label not working

### DISCUSSION THREAD

**Discussion by braini077** [t0bst4r/home-assistant-matter-hub#324] [Q&A]
I just installed build 55 and created a hub.
I want to add devices via label, so I created a label and added them to 2 devices.
The MatterHub will not recognize them. If I include them via pattern, they are added.

I tried 2 different label names.  
Trying to show the labels of an entity shows an empty result:

`{{ labels("switch.esstisch") }}`
Result: [ ]

Maybe a HA issue?? 
Core: 2024.12.3
OS: 14.0

**Comment by k-dani** [comment]
Home-Assistant-Matter-Hub Version: 3.0.0-alpha.55 

For me it's working to add device entities via labels but it's tricky:

You have to label the entities you want to add and not devices. Labeling devices doesn't work (for now? i don't know).
![grafik](https://github.com/user-attachments/assets/87f7e5d7-6d91-4c8e-abb0-b0d9105166ba)

You can querry the label then.
![grafik](https://github.com/user-attachments/assets/57ee760c-3f34-4b2d-834d-98f106e8d244)

I have to restart home assistant to be able to see new devices in my Alexa App. Just restarting the addon isn't sufficient with  this alpha version. But it's working fine then, i have 28 entities in use atm.

![grafik](https://github.com/user-attachments/assets/5d741f15-2aff-464f-908f-d1c606da32ae)

**Reply by k-dani** [reply]
Seems it's not planned that labels on devices are applied to all it's entities regarding this closed HASS Core issue:
https://github.com/home-assistant/core/issues/114625
